### ISSUE TITLE
Consider adding support for content alignment on parent blocks

### ISSUE THREAD

**Issue by mikachan** [WordPress/gutenberg#57346]
## What problem does this address?

I've noticed recently that a common use case is setting a unified content/text alignment on nested blocks. For example, centering all headings and paragraphs in a group block. Currently, the only way to do this is to set the text alignment on each block individually.

## What is your proposed solution?

Allow blocks that are often used as parents (group, column) to set alignment that propagates to any nested blocks. This might need to cover both text alignment and justification settings, depending on the nested blocks, and depending on how many blocks we'd want this to affect.

**Comment by Marc-pi** [comment]
yep, frequent usage, i do confirm the need for whoever builds frequently his theme from the UI.

**Comment by Nyiriland** [comment]
Came here to say I would also gladly use this feature.

**Comment by benoitchantre** [comment]
@Mamaduka I have tried to add textAlign support to the group block. My expectation was to get the related control, but it didn't appear.

```json
"typography": {
	"fontSize": true,
	"lineHeight": true,
	"textAlign": true,
	"__experimentalFontFamily": true,
	"__experimentalFontWeight": true,
	"__experimentalFontStyle": true,
	"__experimentalTextTransform": true,
	"__experimentalTextDecoration": true,
	"__experimentalLetterSpacing": true,
	"__experimentalDefaultControls": {
		"fontSize": true
	}
},
```

https://developer.wordpress.org/block-editor/reference-guides/block-api/block-supports/#typography-textalign

**Comment by Mamaduka** [comment]
@benoitchantre, how are you modifying the `supports` arguments for the Group block? The following example works in my tests.

```php
add_filter( 'register_block_type_args', function( $args, $block_type ) {
	if ( 'core/group' === $block_type ) {
		$args['supports']['typography']['textAlign'] = true;
	}

	return $args;
}, 10, 2 );
```

## Screenshot

![Image](https://github.com/user-attachments/assets/a877d19b-b5f8-4a2b-953b-6c8702b773f6)

**Comment by benoitchantre** [comment]
@Mamaduka Thank you for your feedback. I have directly updated the block.json file, but I don't remember exactly what I did. This was the first time I used the wp-env environment and maybe I missed something during the Contributor Day, as I was also working on a patch for core.

I checked again today and can confirm this is working as expected with the same result as your screenshot 🎉 

What are your thoughts on this issue? That seems to be a nice enhancement for the content editors.

**Comment by Mamaduka** [comment]
Sorry for the late reply, @benoitchantre, I was on vacation.

> What are your thoughts on this issue? That seems to be a nice enhancement for the content editors.

@t-hamano, worked on this feature. He might have more details on why it's not implemented in core for container blocks. Meanwhile, the snippet I shared can be used to enable the feature on the consumer side.

**Comment by t-hamano** [comment]
I don't have a strong opinion on whether to add text alignment support to container blocks. However, the problem is that text alignment cannot be overridden at the block instance level. This is why none of the core blocks support text alignment yet. See https://github.com/WordPress/gutenberg/pull/62260#issuecomment-2257940685 for more details.

To be more specific, if we add support for this to the Group block in the following way:

```diff
diff --git a/packages/block-library/src/group/block.json b/packages/block-library/src/group/block.json
index 3c7d8d3ce1..7fec1aab14 100644
--- a/packages/block-library/src/group/block.json
+++ b/packages/block-library/src/group/block.json
@@ -74,6 +74,7 @@
                        "sticky": true
                },
                "typography": {
+                       "textAlign": true,
                        "fontSize": true,
                        "lineHeight": true,
                        "__experimentalFontFamily": true,
```

If we center text via the global styles and then left-justify text in a block instance, the style won't actually be prioritized.

https://github.com/user-attachments/assets/c85a6edd-0768-452e-a836-474f87629a5b

**Comment by benoitchantre** [comment]
Thank you for your feedbacks and the additional information.
